src/signal.cr

```diff
@@ -24,8 +24,10 @@ require "c/unistd"
 # sleep 3
 # ```
 #
-# Note:
-# - An uncaught exception in a signal handler is a fatal error.
+# NOTE: `Process.on_interrupt` is preferred over `Signal::INT.trap`, as the
+# former also works on Windows.
```

Comment:
We can leave it out of the scope of this PR, but would it be feasible to implement `Signal::INT.trap` as `Process.on_interrupt` on Windows?